### PR TITLE
Add contribution documentation

### PR DESCRIPTION
**Pull request by zkat**
This PR adds a Code of Conduct as well as a contributing guide. Hopefully, this will give folks hoping to contribute a clear idea of what to expect, as well as information on what they can do and how they can do it.

Before I complete this PR, it would be really great to get feedback on the general structure of CONTRIBUTING.md in case I'm just making some grave mistake. Cause it's a lot of writing to be doing :\

### CODE REVIEW

File: CODE_OF_CONDUCT.md
Comment (varjmes) [inline CODE_OF_CONDUCT.md]: "projet" should be "project"

File: CODE_OF_CONDUCT.md
Comment (Charlotteis) [inline CODE_OF_CONDUCT.md]: cute :D

File: CONTRIBUTING.md
Comment (varjmes) [inline CONTRIBUTING.md]: I don't think this sentence is needed at all.

File: CONTRIBUTING.md
Comment (varjmes) [inline CONTRIBUTING.md]: remove "possibly"

File: CONTRIBUTING.md
Comment (varjmes) [inline CONTRIBUTING.md]: Would a bug fix also need to be "COMPONENT"?

**Review by varjmes — CHANGES_REQUESTED** [review]

**Comment by varjmes** [comment]
This is fantastic.
Can I steal these for everything I ever do?

🎶 I have some questions, a couple of suggestions (in my review) 🎶

File: CODE_OF_CONDUCT.md
Comment (zkat) [reply CODE_OF_CONDUCT.md]: it's fine. It's French.

File: CONTRIBUTING.md
Comment (zkat) [reply CONTRIBUTING.md]: I feel like I should clarify this cause it got confusing -- and as far as bugfixes though, I think I'm more interested in the _component_ the commit affects, rather than whether something is a bugfix or a feature tweak or whatever? So `extract: fix tarball lockup issue` rather than `fix: tarball extract issue`.

An alternative is to just go ahead and use angular-style commit messages, which have a standardized format and some automated tooling built around them. See: https://github.com/conventional-changelog/conventional-changelog-angular/blob/master/convention.md

What do you think? I'm not interested in using semantic-release, but this format would actually let us autogenerate changelogs, which would be handy.

**Comment by zkat** [comment]
@Charlotteis thanks for taking the time to review! It sounds like the format is good, so I'll sit down and bang out the rest of the document so we can do one final review before merging.

And as far as using it goes: please, feel free to use any of the docs here for anything you want. It's all public domain and you know how I am about documentation reuse <3

File: CODE_OF_CONDUCT.md
Comment (varjmes) [reply CODE_OF_CONDUCT.md]: No.

Comment (varjmes) [reply CODE_OF_CONDUCT.md]: 🤔

File: CONTRIBUTING.md
Comment (varjmes) [reply CONTRIBUTING.md]: I think I tend to use angular-style. I like that style but having yours clarified, I also understand the component based approach :)

File: CODE_OF_CONDUCT.md
Comment (zkat) [reply CODE_OF_CONDUCT.md]: screen shot 2017-02-16 at 08 25 00

is too

**Review by varjmes — APPROVED** [review]

File: CONTRIBUTING.md
Comment (varjmes) [inline CONTRIBUTING.md]: remove "simply"

File: CONTRIBUTING.md
Comment (varjmes) [inline CONTRIBUTING.md]: remove "simply" x 2

**Comment by zkat** [comment]
@Charlotteis could you review again? This is done.

File: CONTRIBUTING.md
Comment (varjmes) [inline CONTRIBUTING.md]: "approved for merge" sounds better than "approved or merger" to me.

**Review by varjmes — CHANGES_REQUESTED** [review]

**Comment by varjmes** [comment]
One nitpick.

Also, a question: What 'role' do I have? Committer, Labeller, Owner, Admin?
I see I have the tag 'Collaborator' and I have committed, so I am assuming committer with labelling abilities? :) Just want to know so I know what the other roles look like.

**Review by varjmes — APPROVED** [review]

**Comment by varjmes** [comment]
This is great!

Issues for the following sections:
1. Merge Pull Requests
1. Tag a Release

Should be created :)

**Comment by varjmes** [comment]
:shipit:

I also created two issues to fill the remaining two sections.

This is awesome. Thanks for writing this and I look forward to using them in my own projects.

**Pull request merged**
